oem/ibm/libpldmresponder/oem_ibm_handler.cpp

```diff
@@ -749,6 +1010,243 @@ int encodeEventMsg(uint8_t eventType, const std::vector<uint8_t>& eventDataVec,
+        static constexpr auto dumpParam =
+            "com.ibm.Dump.Create.CreateParameters.DumpType";
+        static constexpr auto errLogIdParam =
+            "com.ibm.Dump.Create.CreateParameters.ErrorLogId";
+        static constexpr auto failingUintIdParam =
+            "com.ibm.Dump.Create.CreateParameters.FailingUnitId";
+        static constexpr auto sbeParam = "com.ibm.Dump.Create.DumpType.SBE";
+        static constexpr auto msbeParam =
+            "com.ibm.Dump.Create.DumpType.MemoryBufferSBE";
+
+        std::map<std::string, std::variant<std::string, uint64_t>> createParams;
+        if (entityType == PLDM_ENTITY_MEMORY_MODULE)
+        {
+            createParams[dumpParam] = msbeParam;
+        }
+        else if (entityType == PLDM_ENTITY_PROC)
+        {
+            createParams[dumpParam] = sbeParam;
+        }
+        else
+        {
+            error("Invalid entity type received: {ENTITY_TYPE}", "ENTITY_TYPE",
+                  entityType);
+            return PLDM_ERROR;
+        }
+        createParams[errLogIdParam] = (uint64_t)value;
+        createParams[failingUintIdParam] = (uint64_t)entityInstance;
+        method.append(createParams);
+
+        auto response = bus.call(method);
+
+        sdbusplus::message::object_path reply;
+        response.read(reply);
+
+        monitorDump(reply, entityType, entityInstance);
+    }
+    catch (const std::exception& e)
+    {
+        std::cerr
+            << "Failed to make a DBus call as the dump policy is disabled,ERROR= "
+            << e.what() << "\n";
+        // case when the dump policy is disabled but we set the host effecter as
+        // true and the host moves on
+        setHostEffecterState(true, entityType, entityInstance);
+    }
+    return PLDM_SUCCESS;
+}
+
+std::vector<std::string>
+    pldm::responder::oem_ibm_platform::Handler::getProcObjectPaths()
+{
+    static constexpr auto searchpath = "/xyz/openbmc_project/inventory/system";
+    int depth = 0;
+    std::vector<std::string> procInterface = {
+        "xyz.openbmc_project.Inventory.Item.Cpu"};
+    pldm::utils::GetSubTreeResponse response =
+        dBusIntf->getSubtree(searchpath, depth, procInterface);
+    std::vector<std::string> procPaths;
+    for (const auto& [objPath, serviceMap] : response)
+    {
+        procPaths.emplace_back(objPath);
+    }
+    return procPaths;
+}
+
+std::vector<std::string>
+    pldm::responder::oem_ibm_platform::Handler::getDimmObjectPaths()
+{
+    static constexpr auto searchpath = "/xyz/openbmc_project/inventory/system";
+    static constexpr auto itemPath = "xyz.openbmc_project.Inventory.Item.Dimm";
+    int depth = 0;
+    std::vector<std::string> dimmInterface = {itemPath};
+    pldm::utils::GetSubTreeResponse response =
+        dBusIntf->getSubtree(searchpath, depth, dimmInterface);
+    std::vector<std::string> dimmPaths;
+    for (const auto& [objPath, serviceMap] : response)
```

Comment:
GetSubtreePaths will return array of object paths.  Check https://github.com/openbmc/docs/blob/master/architecture/object-mapper.md